### ISSUE TITLE
Wing panel missing after installation

### ISSUE THREAD

**Issue by Jane2213** [stsdc/monitor#172]
after you install this app for the first time the wing panel hides and you have to open the applications menu or restart to get it back.

**Comment by blchinezu** [comment]
For me wingpanel actually crashes if I try to open the Application menu. I have to manually launch the wingpanel binary.

After a reboot it works fine though.

**Comment by ryonakano** [comment]
Could you open your terminal, run the following command, and paste the output here?

```
killall wingpanel
killall wingpanel
wingpanel
```

**Comment by blchinezu** [comment]
I don't really see the point in that. There are no errors except for this warning which is probably unrelated:
`** (wingpanel:368): WARNING **: 00:32:19.987: Volume-control.vala:476: pa_context_connect() failed: Connection refused`

It did ring a bell though. I could reproduce the failure of starting wingpanel by opening the Apllication Menu through the win+space hotkey. I've monitored the processes and `wingpanel --toggle-indicator=app-launcher` is launched when trying to open the Application Menu. It seems like it crashes with a segault:

```
gabi@983jrfuiwejf98:~$ wingpanel --toggle-indicator=app-launcher

** (wingpanel:368): WARNING **: 00:32:19.987: Volume-control.vala:476: pa_context_connect() failed: Connection refused

Segmentation fault
gabi@983jrfuiwejf98:~$
```

**Comment by ryonakano** [comment]
@blchinezu Hmm, it seems that `Volume-control.vala` which was appeared in your log is part of Sound Indicator and not part of Monitor nor Applications Menu.

Did you install non-official indicators? Also, did you happen to see this issue after you installed Monitor and never faced before?

**Comment by blchinezu** [comment]
> Did you install non-official indicators?

Nope, no other custom indicators of any kind. I keep my OS as clean as possible.

> Also, did you happen to see this issue after you installed Monitor and never faced before?

Now that's something I don't really know. I would've had to kill wingpanel intentionally and then try to launch it through the hotkey. Not something you just do everyday :))

It's not that big of a deal. It only happens right after the install.

A more pressing issue would be the fact that it's crashing once in a while. I'll try to monitor it and see if I can figure something out. For the moment I'm running this to keep it alive:

```bash
#!/bin/bash

APP="com.github.stsdc.monitor"
NAME="System Monitor Indicator"
ICON="utilities-system-monitor"

while [ true ]; do
    notify-send -i $ICON "$NAME" "Launching"
    $APP
    sleep 1s
done
```

Not the best but does the job.

**Comment by blchinezu** [comment]
```bash
...

** (com.github.stsdc.monitor:17656): CRITICAL **: 13:15:05.656: string_contains: assertion 'needle != NULL' failed

** (com.github.stsdc.monitor:17656): CRITICAL **: 13:15:05.656: string_contains: assertion 'needle != NULL' failed

** (com.github.stsdc.monitor:17656): CRITICAL **: 13:15:05.656: string_contains: assertion 'needle != NULL' failed

** (com.github.stsdc.monitor:17656): WARNING **: 13:15:15.683: ProcessUtils.vala:47: Error reading cmdline file '/proc/31450/cmdline': Error opening file /proc/31450/cmdline: No such file or directory

/home/gabi/.bin/system-monitor-indicator.sh: line 12: 17656 Segmentation fault      G_MESSAGES_DEBUG=all com.github.stsdc.monitor 2>&1 >> /tmp/${APP}.log

** (com.github.stsdc.monitor:31486): CRITICAL **: 13:15:17.291: monitor_process_utils_sanitize_commandline: assertion 'commandline != NULL' failed

** (com.github.stsdc.monitor:31486): CRITICAL **: 13:15:17.291: string_contains: assertion 'needle != NULL' failed

** (com.github.stsdc.monitor:31486): CRITICAL **: 13:15:17.291: string_contains: assertion 'needle != NULL' failed

** (com.github.stsdc.monitor:31486): CRITICAL **: 13:15:17.291: string_contains: assertion 'needle != NULL' failed
```

**Comment by stsdc** [comment]
Looks like this error must be handled better:
https://github.com/stsdc/monitor/blob/5b11c95376f452be4e348b408f092a3074767af2/src/Managers/ProcessUtils.vala#L24-L50

Need confirmation if it's really causing this. @blchinezu could You run a few more times to confirm?

I also do have crushes of Wingpanel, but it restarts by itself. And happens rarely.

**Comment by blchinezu** [comment]
wingpanel doesn't crash when that segfault is thrown. Only the monitoring app.

And I've seen the same error multiple times.

**Comment by stsdc** [comment]
Well, when Monitor crashes it loses connections with the Indicator. Indicator hangs, which might cause Wingpanel hang. Might be connected. Create another issue and mention it here.

**Comment by blchinezu** [comment]
Created https://github.com/stsdc/monitor/issues/177

**Comment by stsdc** [comment]
For me, when Monitor crushes, the Indicator just disappears. Without Wingpanel crush.

**Comment by blchinezu** [comment]
Same here. The wingpanel issue (not starting by itself) only happened right after the instalation of the Monitor app. And a reboot or manually launching `wingpanel` solves the install-time issue.

**Comment by stsdc** [comment]
I'm, too,  having Wingpanel crushes, but it restarts itself. and occurs very rarely.

**Comment by stsdc** [comment]
Just restarted my desktop: no crushes when clicking, nor "super + space". Wingpanel crushes are not super rare, they might happened to me even with Monitor Indicator was disable.

May I ask You to try to disable Monitor Indicator and observe Wingpanel?

**Comment by blchinezu** [comment]
Wingpanel is not crashing once I've got Monitor installed though. It is only stopped while installing and it's not automatically starting again. Unfortunately I could not reproduce the install-time issue again, probably because of some already installed dependencies or configs.

Once the monitor app is running and wingpanel is running too, wingpanel is not crashing. Only Monitor randomly stops.

So it's pointless for me to disable Monitor since I don't have wingpanel crashes anyway.

**Comment by stsdc** [comment]
I'm not able to reproduce this. Someone has to gather more info.

Kill and start Wingpanel with a debug messages:
1.`killall wingpanel` 
2.`G_MESSAGES_DEBUG=all wingpanel`
and try to reproduce an issue.

**Comment by marco4mp** [comment]
> I'm not able to reproduce this. Someone has to gather more info.
> 
> Kill and start Wingpanel with a debug messages:
> 1.`killall wingpanel`
> 2.`G_MESSAGES_DEBUG=all wingpanel`
> and try to reproduce an issue.

Hi.
The last line says this:
(wingpanel:2680): GLib-GIO-ERROR **: 20:51:15.148: Settings schema 'com.github.stsdc.monitor.settings' is not installed

**Comment by stsdc** [comment]
Well, looks like monitor was not installed correctly. Try to reinstall. I did complete wipe of Monitor and reinstalled it again. There was no Wingpanel crush.
